hydra/_internal/utils2.py

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
```

Comment:
I created this new file because I was getting circular import issues when I put the `deprecation_warning` function in `hydra.utils` or `hydra.core.utils` or `hydra._internal.utils`.

Reply:
That's odd. can you give more info about the circular import?
what is the cycle?

Reply:
Adding `def deprecation_warning(...)` to `hydra/_internal/utils.py` and putting `from hydra._internal.utils import deprecation_warning` in `hydra/core/utils.py`, we get this cycle:
```text
hydra._internal.utils
-> hydra.core.utils
-> hydra._internal.utils
```

Adding `def deprecation_warning(...)` to `hydra/core/utils.py` and putting `from hydra.core.utils import deprecation_warning` in `hydra/core/default_element.py`, we get this cycle:
```text
hydra.core.utils
-> hydra.core.hydra_config
-> hydra.conf
-> hydra.core.config_store
-> hydra.plugins.config_source
-> hydra.core.default_element
-> hydra.core.utils
```

Adding `def deprecation_warning(...)` to `hydra/utils.py` and putting `from hyd.... import deprecation_warning` in `hydra/core/default_element.py`, we get a similar cycle:
```text
hydra.utils
-> hydra._internal.instantiate._instantiate2
-> hydra._internal.utils
-> hydra.core.utils
-> hydra.core.hydra_config
-> hydra.conf
-> hydra.core.config_store
-> hydra.plugins.config_store
-> hydra.plugins.config_source
-> hydra.core.default_element
-> hydra.utils
```

Reply:
Okay, put it in a module with a better name.

`hydra._internal.deprecation_warning` ?

Reply:
Done in https://github.com/facebookresearch/hydra/pull/1703/commits/801693dff7350dad2883dfc0213ae3843e8c0cb8.